externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/spark/kyuubi/SparkSQLLineageParseHelper.scala

```diff
@@ -0,0 +1,309 @@
+        case (k, attrs) =>
+          k.name -> attrs.map(_.name).toSet
+      }
+    val (inputTables, outputTables) = columnsLineage.foldLeft((List[String](), List[String]())) {
+      case ((inputs, outputs), (out, in)) =>
+        val x = (inputs ++ in.map(_.split('.').init.mkString("."))).filter(_.nonEmpty)
+        val y = outputs ++ List(out.split('.').init.mkString(".")).filter(_.nonEmpty)
+        (x, y)
+    }
+    Lineage(inputTables.distinct, outputTables.distinct, columnsLineage)
+  }
+
+  private def mergeColumnsLineage(
+      left: AttributeMap[AttributeSet],
+      right: AttributeMap[AttributeSet]): AttributeMap[AttributeSet] = {
+    left ++ right.map {
+      case (k, attrs) =>
+        k -> (attrs ++ left.getOrElse(k, AttributeSet.empty))
+    }
+  }
+
+  private def joinColumnsLineage(
+      parent: AttributeMap[AttributeSet],
+      child: AttributeMap[AttributeSet]): AttributeMap[AttributeSet] = {
+    if (parent.isEmpty) child
+    else {
+      val childMap = child.map { case (k, attrs) => (k.exprId, attrs) }
+      parent.map { case (k, attrs) =>
+        k -> AttributeSet(attrs.flatMap(attr =>
+          childMap.getOrElse(attr.exprId, AttributeSet.empty)))
+      }
+    }
+  }
+
+  private def getSelectColumnLineage(
+      named: Seq[NamedExpression]): AttributeMap[AttributeSet] = {
+    val exps = named.map {
+      case a: Alias => a.toAttribute -> a.references
+      case a: Attribute => a -> a.references
+    }
+    ListMap(exps: _*)
+  }
+
+  private def joinRelationColumnLineage(
+      parent: AttributeMap[AttributeSet],
+      relationAttrs: Seq[Attribute],
+      tableName: String = ""): AttributeMap[AttributeSet] = {
+    val relationAttrSet = AttributeSet(relationAttrs)
+    if (parent.nonEmpty) {
+      parent.map { case (k, attrs) =>
+        k -> AttributeSet(attrs.collect {
+          case attr if relationAttrSet.contains(attr) =>
+            attr.withName(Seq(tableName, attr.name).filter(_.nonEmpty).mkString("."))
+        })
+      }
+    } else {
+      ListMap(relationAttrs.map { attr =>
+        (
+          attr,
+          AttributeSet(attr.withName(Seq(tableName, attr.name).filter(_.nonEmpty).mkString("."))))
+      }: _*)
+    }
+  }
+
+  private def extractColumnsLineage(
+      plan: LogicalPlan,
+      parentColumnsLineage: AttributeMap[AttributeSet]): AttributeMap[AttributeSet] = {
+
+    def getPlanField[T](field: String): T = {
+      getFieldVal[T](plan, field)
+    }
+
+    def getCurrentPlanField[T](curPlan: LogicalPlan, field: String): T = {
+      getFieldVal[T](curPlan, field)
+    }
+
+    def getPlanMethod[T](name: String): T = {
+      getMethod[T](plan, name)
+    }
+
+    def getQuery: LogicalPlan = {
+      getPlanField[LogicalPlan]("query")
+    }
+
+    plan match {
+      // For command
+      case p if p.nodeName == "AlterViewAsCommand" =>
+        val analyzed = sparkSession.sessionState.analyzer.execute(getQuery)
```

Comment:
why we need this ?

Reply:
In Spark3.1, the subquery plan of `AlterViewAsCommand`  is not unresolved. In addition, Spark 3.2/3.3 has been resolved

Reply:
oh, I see